smithy-jsonschema/src/main/java/software/amazon/smithy/jsonschema/DefaultRefStrategy.java

```diff
@@ -130,6 +135,11 @@ public boolean isInlined(Shape shape) {
             return isInlined(target);
         }
 
+        // smithy.api#Unit is always inlined
+        if (UnitTypeTrait.UNIT.equals(shape.getId())) {
```

Comment:
I think things like API Gateway will generate names for anonymous objects (which this code would cause). Is there a way to generate a Unit shape in the JSON Schema for tagged unions members, but not for operation input?

Reply:
I have hacked this PR to do this, but the implementation via a boolean flag screams "this is not the right way to accomplish this."

This would be so much easier if we had service attached shapes; then I could meaningfully use relationship data to examine what's actually attached to the scope of the conversion, and a lot of this code would be a lot simpler.